File: C#/RefactorMe/ProductDataConsolidator.cs
using RefactorMe.DontRefactor.Data.Implementation;
using RefactorMe.DontRefactor.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RefactorMe.Enum;

namespace RefactorMe
{
    public class ProductDataConsolidator
    {
        /// <summary>
        /// Get list of products with unconverted price currency
        /// </summary>
        /// <returns>List of <see cref="Product"/></returns>
        public static List<Product> Get() {
            return GenerateProductList();
        }

        /// <summary>
        /// Get list of products with converted price currency to Euro
        /// </summary>
        /// <returns>List of <see cref="Product"/></returns>
        public static List<Product> GetInEuros() {
            return GenerateProductList(0.67);
        }

        /// <summary>
        /// Get list of products with converted price currency to Dollar
        /// </summary>
        /// <returns>List of <see cref="Product"/></returns>
        public static List<Product> GetInUSDollars()
        {
            return GenerateProductList(0.76);
        }

        /// <summary>
        /// Generates a list of products from multiple repository data
        /// </summary>
        /// <param name="currencyMultiplier">Currency multiplier: Price is multiplied with the paramater, if not provided defaults to 1</param>
        /// <returns>List of <see cref="Product"/></returns>
        private static List<Product> GenerateProductList(double currencyMultiplier = 1)
        {
            IQueryable<Lawnmower> lawnmowers = new LawnmowerRepository().GetAll();
            IQueryable<PhoneCase> phoneCases = new PhoneCaseRepository().GetAll();
            IQueryable<TShirt> tShirts = new TShirtRepository().GetAll();

            var products = new List<Product>();
            products.AddRange(
                lawnmowers.Select(x => new Product
                {
                    Id = x.Id,
                    Name = x.Name,
                    Price = x.Price * currencyMultiplier,
                    Type = EnumItemType.Lawnmower.ToString()
                })
                .Concat(
                    phoneCases.Select(x => new Product
                    {
                        Id = x.Id,
                        Name = x.Name,
                        Price = x.Price * currencyMultiplier,
                        Type = EnumItemType.PhoneCase.ToString()
                    })
                ).Concat(
                    tShirts.Select(x => new Product
                    {
                        Id = x.Id,
                        Name = x.Name,
                        Price = x.Price * currencyMultiplier,
                        Type = EnumItemType.TShirt.ToString()
                    })
                ));

            return products;
        }
    }
}

File: JavaScript/src/refactor-me/ProductDataConsolidator.js
function ProductDataConsolidator() { }

ProductDataConsolidator.get = function () {
	return ProductDataConsolidator.generateProductList();
}

ProductDataConsolidator.getInUSDollars = function () {
	return ProductDataConsolidator.generateProductList(0.76);
}

ProductDataConsolidator.getInEuros = function () {
	return ProductDataConsolidator.generateProductList(0.67);
}

ProductDataConsolidator.generateProductList = function (currencyMultiplier = 1) {
	let lawnmowers = new LawnmowerRepository().getAll();
	let phoneCases = new PhoneCaseRepository().getAll();
	let tShirts = new TShirtRepository().getAll();
	let products = [];

	lawnmowers.forEach(item => {
		products.push({
			id: item.id,
			name: item.name,
			price: item.price * currencyMultiplier,
			type: "Lawnmower"
		});
	});
	phoneCases.forEach(item => {
		products.push({
			id: item.id,
			name: item.name,
			price: item.price * currencyMultiplier,
			type: "Phone Case"
		});
	});
	tShirts.forEach(item => {
		products.push({
			id: item.id,
			name: item.name,
			price: item.price * currencyMultiplier,
			type: "T-Shirt"
		});
	});

	return products;
}
